src/Services/Ordering/Ordering.API/Application/Commands/CreateOrderCommandHandler.cs

```diff
@@ -46,12 +48,19 @@ public async Task<bool> Handle(CreateOrderCommand message, CancellationToken can
             // methods and constructor so validations, invariants and business logic 
             // make sure that consistency is preserved across the whole aggregate
             var address = new Address(message.Street, message.City, message.State, message.Country, message.ZipCode);
-            var order = new Order(message.UserId, message.UserName, address, message.CardTypeId, message.CardNumber, message.CardSecurityNumber, message.CardHolderName, message.CardExpiration);
+            var orderItems = message.OrderItems.Select(item => new OrderItem(item.ProductId, item.ProductName,
+                item.UnitPrice, item.Discount, item.PictureUrl, item.Units));
 
-            foreach (var item in message.OrderItems)
-            {
-                order.AddOrderItem(item.ProductId, item.ProductName, item.UnitPrice, item.Discount, item.PictureUrl, item.Units);
-            }
+            var order = new Order(
```

Comment:
If you wish to handle bulk complexity I would rather opt into implementing `AddOrderItems` instead of CTOR where you lose some flexibility like returning results, async, etc..

Reasons being AddOrderItem can/could change from `void` to some functional object, validation, error result with some success flag for instance since it should encapsulate some logic which currently is not there but this is still a showcase of the pattern.